simulation/traffic_simulator/include/traffic_simulator/entity/ego_entity.hpp

```diff
@@ -41,6 +41,11 @@ class EgoEntity : public VehicleEntity
 
   CanonicalizedEntityStatus externally_updated_status_;
 
+  bool is_controlled_by_simulator_{false};
+  std::optional<double> target_speed_;
+  traffic_simulator_msgs::msg::BehaviorParameter behavior_parameter_;
+  std::shared_ptr<traffic_simulator_msgs::msg::PolylineTrajectory> polyline_trajectory_;
```

Comment:
I believe that this would introduce ambiguity in the use of these variables by the other classes (`Vehicle`, `Pedestrian`..). In these classes there also occurs modification of `BehaviorParameter`, `target_speed`, etc., but these values are stored and modified directly in `BehaviorTree` - there is no need to store them on the entities class objects side. 
For `EgoEntity` this is necessary, as `BehaviorTree` is not used. My opinion is that if we want to move these objects to `EntityBase`, we should think about it more deeply in the context of how these variables are modified and used.

Obviously, if you think that anyway it is good idea to move these variables now to `EntityBase` let me know - it is possible that there is some use for them that I missed during my considerations or you have knowledge that in the future there will be a need to use them on the side of entity classes.

Reply:
OK, I understand.
I withdraw the change request.